### ISSUE TITLE
build_runner test sourcemaps are ignored

### ISSUE THREAD

**Issue by daniel-v** [dart-lang/build#2901]
**Short summary:**

Running `pub run build_runner test -r -- -p chrome` command, I'd expect Dart stack traces to be reported in console. It is not the case.

**Things I tried**

In build yaml, I have the following to ensure that after building the assets, I retain the sourcemaps.

```yaml
global_options:
  build_web_compilers|dart_source_cleanup:
    release_options:
      enabled: false
  build_web_compilers|dart2js_archive_extractor:
    release_options:
      filter_outputs: false
```

I verified that they are in fact loaded, by starting tests with `--pause-after-load` and opening a dart file in chrome's dev toolbar. I'm not sure where the issue is but:

- I have sourcemaps
- they get loaded in browser
- I see non-mapped JS stack trace in console for errors

Anything I can do to get Dart traces?

Extra info:

* Dart SDK version: 2.9.3 (stable)
```
  build_runner:
    version: "1.9.0"
  build_runner_core:
    version: "5.1.0"
  build_test:
    version: "1.0.0"
  build_web_compilers:
    version: "2.10.0"
```
* Using it for an Angular 5.3.1 app

This issue might be related to https://github.com/dart-lang/build/issues/2377 but I'm not sure.

**Comment by jakemac53** [comment]
I can repro this at HEAD, it looks like we aren't cleaning up the uris in the source maps like we need to, so they have `org-dartlang-app` scheme uris.

**Comment by jakemac53** [comment]
I can fix this issue when running the test manually, but it looks like we don't support stack trace mapping for dart2js the way the test package wants it set up (we need to register a stack trace mapper on the page I think, which will be a fair bit of work given the current setup). I am not sure dart2js actually supports it either like DDC does.

We may be able to register a standard stack trace mapper on the package:test side of things, when in precompiled mode, which we don't do today.

**Comment by daniel-v** [comment]
As far as I can remember, the `*_test.browser.dart` files (might not remember the exact extension) wraps each of the tests in some code (channel back to the test runner eg.). Most importantly, I saw some code related to a mapper. First glance, it seemed that there is a mapper registered on client side which attempts to map stack traces around serialization. 

Not sure if this helps but figured I'd drop it here if you were so quick to jump on this issue.

**Extra context info**: currently, we compile 2x on CI. Once, so that we get human readable stack traces and once to create the release artifact. Therefore, it takes 2x as long to do a test-build run. Also, we'd much prefer running tests with release flags (`-r -O3 --fast-startup etc`) to test the code that will be ran in the browser.

**Comment by daniel-v** [comment]
Wow, @jakemac53 I really appreciate the quick fix. I'm super happy about it.

**Comment by jakemac53** [comment]
https://github.com/dart-lang/build/pull/2903 will at least allow you to manually run the tests - using `pub run build_runner serve test -r` and then loading up `https://localhost:8080` in your browser and selecting the test.

The console in the browser still output isn't helpful (its the same as what you see piped through when you run the test command), but you can tell it to pause on caught errors, and there the source maps will start working.

Going to leave this issue open still because we would like to also get the stack trace mapping working for console output.

**Comment by jakemac53** [comment]
> As far as I can remember, the `*_test.browser.dart` files (might not remember the exact extension) wraps each of the tests in some code (channel back to the test runner eg.). Most importantly, I saw some code related to a mapper. First glance, it seemed that there is a mapper registered on client side which attempts to map stack traces around serialization.

Ya this asks for a "mapper" from the test runner channel communicating to the browser if I recall correctly. But for precompiled tests (which build_runner tests are), it doesn't set up a mapper because for DDC at least build_runner does its own custom mapping. It however does not do this for dart2js, which is I believe the root cause here.

**Comment by jakemac53** [comment]
I just released version `2.15.2` of `build_web_compilers`, which should at least enable local debugging until we can fix the root cause here.
